interview-questions.md

```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: The variables that are used when a funcition is created are know as the parameters. The data or values that are used when the function is called are known as arguments.
+
+answer via:https://byjus.com/gate/difference-between-argument-and-parameter-in-c-and-c-plus-plus/#:~:text=1-,The%20values%20that%20are%20declared%20within%20a%20function%20when%20the,declared%20are%20known%20as%20parameters.&text=These%20are%20used%20in%20function,function%20to%20the%20receiving%20function.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: The .map() method can used 3 different predefined parameters. Those parametes are value, array, and index. For the .map method to work there is only one required parameter which is value the other two are optional.
+
+Researched answer: There are 3 possible designated parameters for the .map() method. The first is value, this parameter is the value the method is processing at different steps in the function. The next is index. This is simply the index of the current value. Last is array, which is a parameter that points to the original array. A combination of all three parameters can be used however only value is required to use .map().
 
-Researched answer:
+answer via: https://stackabuse.com/guide-to-javascripts-map-method/
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: Map is a method that will take in an array and create a new array after iterating over the old. We can take an array of all lower case strings and use the .map() method to change them to all uppercase because of its ability to look at each value and make the change. Filter is a method which can be used to extract personlized values from an array. For example if we have an array of 100 numbers we can use .filter() to set a conditional statement that will return only odd numbers. 
+
+Researched answer: the method .map() returns an array of informatin from the original array. We can use it in the call back function to change or get only the data wanted. The .filter() method is used to aquire a subset of data from the original array based on provided condition. 
+
+answer via: https://www.google.com/search?q=What+is+the+difference+between+map+and+filter%3F&oq=What+is+the+difference+between+map+and+filter%3F&aqs=chrome..69i57j0i512j0i20i263i512j0i512l2j0i22i30l5.644j0j15&sourceid=chrome&ie=UTF-8
+
+4. What is iteration?
 
-Researched answer:
+Your answer: iteration is an action being performed continuoulsy until the goal is reached. For example in programming there are methods that can be used to accomplish iteration. The .map() method will iterate over an array until all index values have been accounted for. 
 
-1. What is iteration?
+Researched answer: a sequence of code that is repeated until the desired result is completed.
```

Comment:
Good Job! I would encourage you to use your research to explore more into the topic of iteration. See what new things you learn.